datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -257,7 +257,50 @@ impl PhysicalExpr for BinaryExpr {
     fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
         use arrow::compute::kernels::numeric::*;
 
+        fn check_short_circuit(arg: &ColumnarValue, op: &Operator) -> bool {
+            let data_type = arg.data_type();
+            match (data_type, op) {
+                (DataType::Boolean, Operator::And) => {
+                    match arg {
+                        ColumnarValue::Array(array) => {
+                            if let Ok(array) = as_boolean_array(&array) {
+                                return array.true_count() == 0;
```

Comment:
FWIW this `true_count` operation requires checking all values (which should still be fast, but I wanted to point out it was not free / constant time): https://docs.rs/arrow-array/52.0.0/src/arrow_array/array/boolean_array.rs.html#142

Reply:
We probably can speed it up by providing a  function that doesn't count but provides whether the bitmap is all true / all false 🤔